src/storage/mod.rs

```diff
@@ -1421,7 +1426,7 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
         const CMD: CommandKind = CommandKind::raw_batch_get;
         let priority = ctx.get_priority();
         let priority_tag = get_priority_tag(priority);
-        let resource_tag = ResourceMeteringTag::from_rpc_context(&ctx);
+        let resource_tag = self.tag_factory.new_tag(&ctx);
```

Comment:
maybe call it `resource_tag_factory` is better, as we have so many "tags".  A single "tag_factory" will cause ambiguity.